File: LeetCode Solutions/Problems/2180. Count Integers With Even Digit Sum.cpp
class Solution {
public:
    int countEven(int num) {
        int sum = 0, n = num;
        for (sum = 0; num > 0; sum += num % 10, num /= 10);
        return (sum & 1) ? (n - 1) / 2: n / 2;
    }
};

File: LeetCode Solutions/Problems/2180. Count Integers With Even Digit Sum.py
class Solution:
    def countEven(self, num: int) -> int:
        count = 0
        for i in range(2, num+1):
            temp = []
            while i > 0:
                temp.append(i % 10)
                i //= 10
            if sum(temp) % 2 == 0:
                count += 1
        return count